aeppl/transforms.py

```diff
@@ -47,50 +51,57 @@ def class_transformed_rv(op, node):
         return cls_res
 
 
-class RVTransform(abc.ABC):
-    @abc.abstractmethod
-    def forward(self, value: TensorVariable, *inputs: Variable) -> TensorVariable:
-        """Apply the transformation."""
+class TransformedRV(RandomVariable, metaclass=TransformedRVMeta):
+    r"""A base class for transformed `RandomVariable`\s."""
 
-    @abc.abstractmethod
-    def backward(self, value: TensorVariable, *inputs: Variable) -> TensorVariable:
-        """Invert the transformation."""
 
-    def log_jac_det(self, value: TensorVariable, *inputs) -> TensorVariable:
-        """Construct the log of the absolute value of the Jacobian determinant."""
-        # jac = at.reshape(
-        #     gradient(at.sum(self.backward(value, *inputs)), [value]), value.shape
-        # )
-        # return at.log(at.abs_(jac))
-        phi_inv = self.backward(value, *inputs)
-        return at.log(at.nlinalg.det(at.atleast_2d(jacobian(phi_inv, [value]))))
+@singledispatch
+def _default_transformed_rv(
+    op: Op,
+    node: Node,
+) -> Optional[TensorVariable]:
+    """Create a graph for a transformed log-probability of a ``RandomVariable``.
 
+    This function dispatches on the type of ``op``, which should be a subclass
+    of ``RandomVariable``.  If you want to implement new transforms for a
+    ``RandomVariable``, register a function on this dispatcher.
 
-class TransformedRV(RandomVariable, metaclass=DistributionMeta):
-    r"""A base class for transformed `RandomVariable`\s."""
+    """
+    return None
 
 
-@_logprob.register(TransformedRV)
-def transformed_logprob(op, value, *inputs, name=None, **kwargs):
-    """
-    Compute logp graph for a value variable that was back-transformed to be on
-    the natural support of the respective random variable.
-    """
+def create_default_transformed_rv_op(
+    rv_op: Op,
+    transform: Transform,
+    cls_dict_extra: Optional[Dict] = None,
+) -> Type[TransformedRV]:
 
-    logprob = _logprob(op.base_op, value, *inputs, name=name, **kwargs)
+    trans_name = getattr(transform, "name", "transformed")
+    rv_type_name = type(rv_op).__name__
+    cls_dict = type(rv_op).__dict__.copy()
+    rv_name = cls_dict.get("name", "")
+    if rv_name:
+        cls_dict["name"] = f"{rv_name}_{trans_name}"
+    cls_dict["base_op"] = rv_op
+    cls_dict["transform"] = transform
 
-    original_forward_value = op.transform.forward(value, *inputs)
-    jacobian = op.transform.log_jac_det(original_forward_value, *inputs)
+    if cls_dict_extra is not None:
+        cls_dict.update(cls_dict_extra)
 
-    if name:
-        logprob.name = f"{name}_logprob"
-        jacobian.name = f"{name}_logprob_jac"
+    new_op_type = type(f"Transformed{rv_type_name}", (TransformedRV,), cls_dict)
 
-    return logprob + jacobian
+    return new_op_type
+
+
+# =============
+# Transform Opt
+# =============
```

Comment:
Comment-based sectional delimiters are just a personal style choice, and not something that particularly helps the readability/navigation of the code, so let's not add these sorts of things.

Reply:
I'll remove them